File: server/src/main/java/gg/mineral/server/network/packet/login/clientbound/LoginSuccessPacket.java
package gg.mineral.server.network.packet.login.clientbound;

import java.util.UUID;

import com.eatthepath.uuid.FastUUID;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record LoginSuccessPacket(UUID uuid, String username) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeString(os, FastUUID.toString(uuid), username);
    }

    @Override
    public byte getId() {
        return 0x02;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/login/clientbound/LoginSuccessPacket.kt
package gg.mineral.server.network.packet.login.clientbound

import com.eatthepath.uuid.FastUUID
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf
import java.util.*

@JvmRecord
data class LoginSuccessPacket(val uuid: UUID, val username: String) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) =
        os.writeString(FastUUID.toString(uuid), username)

    override val id: Byte
        get() = 0x02
}
